extensions-contrib/prometheus-emitter/src/main/java/org/apache/druid/emitter/prometheus/PrometheusEmitter.java

```diff
@@ -78,9 +81,24 @@ public void start()
         log.error("HTTPServer is already started");
       }
     } else if (strategy.equals(PrometheusEmitterConfig.Strategy.pushgateway)) {
-      pushGateway = new PushGateway(config.getPushGatewayAddress());
+      String address = config.getPushGatewayAddress();
+      if (address.startsWith("https") || address.startsWith("http")) {
+        URL myURL = createURLSneakily(address);
+        pushGateway = new PushGateway(myURL);
+      } else {
+        pushGateway = new PushGateway(address);
+      }
     }
+  }
 
+  private static URL createURLSneakily(final String urlString)
+  {
+    try {
+      return new URL(urlString);
+    }
+    catch (MalformedURLException e) {
+      throw new RuntimeException(e);
```

Comment:
Druid has an "RE" class that does auto-formatting.  Just to make sure you get some information about what's going on here, I'd suggest you actually 

```
throw new RE(e, "couldn't parse https URL from[%s]", urlString);
```